### PR TITLE
Thomson fitting example notebook

### PR DESCRIPTION
**Pull request by pheuer**
Creates an example ipynb for the Thomson fitting functionality

Based on and utilizes #974 , so merge that first

### CODE REVIEW

**Review by rocco8773 — CHANGES_REQUESTED** [review]
Can you add the notebook to the `.. nbgallery:` directive in `docs/ad/diagnostics/thomson.rst`?

**Comment by pheuer** [comment]
> Can you add the notebook to the `.. nbgallery:` directive in `docs/ad/diagnostics/thomson.rst`?

Done 

Note that the codespell error is expected (catching a couple variable names that are not accidental)

**Comment by pheuer** [comment]
@StanczakDominik Thanks - good idea to try but it looks like `ans` is still in the list of potentially misspelled words

```
./docs/notebooks/diagnostics/thomson_fitting.ipynb:412: ans ==> and
./docs/notebooks/diagnostics/thomson_fitting.ipynb:413: ans ==> and
./docs/notebooks/diagnostics/thomson_fitting.ipynb:538: ans ==> and
./docs/notebooks/diagnostics/thomson_fitting.ipynb:539: ans ==> and
```

**Comment by StanczakDominik** [comment]
Just add it to the list of common "typos" in `setup.cfg`, then? It should be under `[codespell]`.

**Comment by pheuer** [comment]
> Just add it to the list of common "typos" in `setup.cfg`, then? It should be under `[codespell]`.

Ooh that works, thanks!! 

I notice that all my charged particle radiography notebooks have been excluded from codespell - does that mean my spelling is beyond reproach or so awful as to not be worth fixing? ;) 

```
*charged_particle*.ipynb
```

**Comment by StanczakDominik** [comment]
Oh LMAO, you're right! `git blame` identifies @namurphy's [016ea25](https://github.com/PlasmaPy/PlasmaPy/pull/1611/commits/016ea25ecc2313d92c89f7c03e5b8b207f8f60fc) as the culprit commit, where it's attached without explanation. What happened there? 😆

**Comment by pheuer** [comment]
My guess is my notebooks contain so many spelling errors it was easiest just to skip them for now haha

**Comment by StanczakDominik** [comment]
It's weird, I removed them from `skip` locally and I'm getting errors on words that I can't find in the actual notebooks. 😕

**Comment by namurphy** [comment]
> Oh LMAO, you're right! `git blame` identifies @namurphy's [016ea25](https://github.com/PlasmaPy/PlasmaPy/pull/1611/commits/016ea25ecc2313d92c89f7c03e5b8b207f8f60fc) as the culprit commit, where it's attached without explanation. What happened there? 😆

If I'm remembering correctly, it was because the notebooks were pre-executed, and codespell was finding "misspellings" in the embedded png files.

**Comment by pheuer** [comment]
> If I'm remembering correctly, it was because the notebooks were pre-executed, and codespell was finding "misspellings" in the embedded png files.

Oh, that makes sense!

**Comment by StanczakDominik** [comment]
OoooooooOOOOOOOOH. That also explains why I couldn't grep out those phrases in the version I was editing via jupytext - it just gives me the text input.

**Comment by pheuer** [comment]
@rocco8773 @namurphy This PR has been in the review phase for a long time. I suggest we either 1) do the review this week or 2) do a light review for major issues and then merge this and fix minor issues later. My concern is that this Thomson fitting functionality is pretty complicated is therefore having it in the package is functionally useless without this example notebook to explain how it works.

File: docs/notebooks/diagnostics/thomson_fitting.ipynb [hunk 1]
```diff
@@ -0,0 +1,739 @@
...
+    "\n",
+    "<img src=\"collective_vs_noncollective_ots.png\">\n",
+    "\n",
+    "Thomson scattering can be either non-collective (dominated by single electron scattering) or collective (dominated by scattering off of electron plasma waves (EPW) and ion acoustic waves (IAW). In the non-collective regime, the scattering spectrum contains a single peak. However, in the collective regime the spectrum contains separate features caused by the electron and ion populations (corresponding to separate scattering off of EPW and IAW). These features exist on different scales: the EPW feature is dim but covers a wide wavelength range, while the IAW feature is bright but narrow. They also encode partially-degenerate information (eg. the flow velocities of the electrons and ions respectively). The two features are therefore often recorded on separate spectrometers and are fit separately. \n"
```
Comment (namurphy) [inline docs/notebooks/diagnostics/thomson_fitting.ipynb:21]:
```suggestion
    "Thomson scattering can be either non-collective (dominated by single electron scattering) or collective (dominated by scattering off of electron plasma waves (EPW) and ion acoustic waves (IAW). In the non-collective regime, the scattering spectrum contains a single peak. However, in the collective regime the spectrum contains separate features caused by the electron and ion populations (corresponding to separate scattering off of EPW and IAW). These features exist on different scales: the EPW feature is dim but covers a wide wavelength range, while the IAW feature is bright but narrow. They also encode partially-degenerate information (e.g., the flow velocities of the electrons and ions respectively). The two features are therefore often recorded on separate spectrometers and are fit separately. \n"
```

**Review by namurphy — COMMENTED** [review]
Thanks for the reminder, @pheuer! I read through the notebook and, with the caveat that I know very little about Thomson scattering diagnostics, it looks pretty close to ready to merge.  @rocco8773 — would you have a chance to look through this soon too?

**Review by namurphy — APPROVED** [review]

**Comment by namurphy** [comment]
@pheuer — Is there anything else you wanted to do before we merged this?  If not, please feel free to merge it, or I could do so soon!

**Comment by pheuer** [comment]
Nope, let's do it!

**Pull request merged**